packages/plugin-ext/src/common/plugin-identifiers.ts

```diff
@@ -81,4 +81,31 @@ export namespace PluginIdentifiers {
         }
         return { id: probablyId.slice(0, endOfName) as UnversionedId, version: probablyId.slice(endOfName + 1) };
     }
+
+    const EXTENSION_IDENTIFIER_WITH_VERSION_REGEX = /^([^.]+\..+)@((prerelease)|(\d+\.\d+\.\d+(-.*)?))$/;
```

Comment:
This is unnecessarily restrictve, IMO: all we need to know here is what is before the '@' and what is after the '@'. Why introduce the added complexity? Also, regexes should always have a comment that describes what they are _supposed_ to detect, Because no-one understands Regexes at first sight. If the id does not match the expression, you end up with the exact same result as before.

Reply:
This is the same RegEx that is also used in vs-code, I thought it is a good idea to do the same.
I will add a comment to describe what is expected by this RegEx.